crates/fuel-core/src/state/rocks_db.rs

```diff
@@ -712,20 +748,36 @@ where
     }
 
     fn get(&self, key: &[u8], column: Self::Column) -> StorageResult<Option<Value>> {
-        self.metrics.read_meter.inc();
-        let column_metrics = self.metrics.columns_read_statistic.get(&column.id());
-        column_metrics.map(|metric| metric.inc());
-
-        let value = self
+        let result = self
             .db
             .get_cf_opt(&self.cf(column), key, &self.read_options)
-            .map_err(|e| DatabaseError::Other(e.into()))?;
+            .map_err(|e| StorageError::Other(DatabaseError::Other(e.into()).into()));
 
-        if let Some(value) = &value {
-            self.metrics.bytes_read.inc_by(value.len() as u64);
-        }
+        self.register_read(result, column.id())
+            .map(|opt| opt.map(Arc::new))
+    }
+
+    fn get_batch<'a>(
+        &'a self,
+        keys: BoxedIter<'a, Vec<u8>>,
+        column: Self::Column,
+    ) -> BoxedIter<'a, StorageResult<Option<Value>>> {
+        let column_family = self.cf(column);
+        let keys = keys.map(|key| (&column_family, key));
 
-        Ok(value.map(Arc::new))
+        self.db
+            .multi_get_cf_opt(keys, &self.read_options)
+            .into_iter()
+            .map(move |result| {
+                self.register_read(
```

Comment:
I like the old approach more, because we called `self.metrics.columns_read_statistic.get(&column)` once, while in you case you will call it for each key, that could be expensive since it uses a lock inside.


<img width="894" alt="image" src="https://github.com/user-attachments/assets/839930b2-9ef6-4bb4-a8f7-596b06916208">

Reply:
Ah I see. I'll revert to the old solution then. I missed the other `multi_get` implementation that already had metrics when I implemented this.

Reply:
Reverted in https://github.com/FuelLabs/fuel-core/pull/2396/commits/f6977073d73ff04110a102bf5a972b4b3a71c316